### DISCUSSION TITLE
Reverse direction M.2 NGFF adapter recommedation

### DISCUSSION THREAD

**Discussion by Jeffrey2019** [osy86/HaC-Mini#561] [General]
![m2 ngff](https://user-images.githubusercontent.com/32722458/74707995-449c9900-5256-11ea-947e-1a9a1651256a.png)

available on [taobao](https://item.taobao.com/item.htm?spm=a1z0d.6639537.1997196601.13.453374842wxbYJ&id=606077440886)

**Comment by chrisallen** [comment]
I ordered one of these, along with 2 IPEX4 extensions, and 1x IPEX4 to an antenna. I will be using a BCM943602CS WiFi card. I can provide install photos once the parts arrive from TBFOCUS.com if you're interested

**Comment by aescolastico** [comment]
Alternatively, here's an m.2 card with the macOS supported chip
[ https://ebay.us/nmM3HK](url)

**Comment by goodbest** [comment]
You can find lots of related discussions here:
https://github.com/osy86/HaC-Mini/issues/86
https://github.com/osy86/HaC-Mini/issues/140

**Comment by chrisallen** [comment]
> Alternatively, here's an m.2 card with the macOS supported chip
> [ https://ebay.us/nmM3HK](url)

I wish I found out about this before ordering a range of parts! :)

**Comment by crhendo** [comment]
> I wish I found out about this before ordering a range of parts! :)

@chrisallen I feel your pain! I'm expecting a number of "try and see" parts from China in the next week or so. Clearly this new part was built for purpose as wherever I look it is quite expensive.

**Comment by we11adam** [comment]
> Alternatively, here's an m.2 card with the macOS supported chip
> [ https://ebay.us/nmM3HK](url)

Have you got one and tested it yet, please? This looks so promising and exciting!

**Comment by aescolastico** [comment]
> > Alternatively, here's an m.2 card with the macOS supported chip
> > [ https://ebay.us/nmM3HK](url)
> 
> Have you got one and tested it yet, please? This looks so promising and exciting!

Yes. Its just a broadcom manufactured chipset on a custom m.2 pcb. To macOS its all the same.

**Comment by jasanders** [comment]
If I swapped from DW1820A to the linked card from eBay, is there a software change I need to make?

**Comment by btbxbob** [comment]
It works!
attach my photo here.
Adjusting the Athena is a little tricky.

![WechatIMG3](https://user-images.githubusercontent.com/2234266/79677392-58975380-8223-11ea-87f5-42986de3da71.jpeg)
![WechatIMG4](https://user-images.githubusercontent.com/2234266/79677394-5a611700-8223-11ea-8500-249057381bfe.jpeg)
![WechatIMG6](https://user-images.githubusercontent.com/2234266/79677395-5af9ad80-8223-11ea-8cb7-30617abe77a4.jpeg)

**Comment by testkam2018** [comment]
> Alternatively, here's an m.2 card with the macOS supported chip
> [ https://ebay.us/nmM3HK](url)

could you please provide the card name? link is dead :(

**Comment by jasanders** [comment]
> > Alternatively, here's an m.2 card with the macOS supported chip
> > [ https://ebay.us/nmM3HK](url)
> 
> could you please provide the card name? link is dead :(

https://www.ebay.com/itm/Broadcom-BCM94360NG-WiFi-Card-better-than-DW1560-BT4-0-Adapter-802-11ac-1200Mbps/283783627652

**Comment by testkam2018** [comment]
> > > Alternatively, here's an m.2 card with the macOS supported chip
> > > [ https://ebay.us/nmM3HK](url)
> > 
> > 
> > could you please provide the card name? link is dead :(
> 
> https://www.ebay.com/itm/Broadcom-BCM94360NG-WiFi-Card-better-than-DW1560-BT4-0-Adapter-802-11ac-1200Mbps/283783627652

thanks! i bought it

**Comment by keatliang2005** [comment]
> It works!
> attach my photo here.
> Adjusting the Athena is a little tricky.
> 
> ![WechatIMG3](https://user-images.githubusercontent.com/2234266/79677392-58975380-8223-11ea-87f5-42986de3da71.jpeg)
> ![WechatIMG4](https://user-images.githubusercontent.com/2234266/79677394-5a611700-8223-11ea-8500-249057381bfe.jpeg)
> ![WechatIMG6](https://user-images.githubusercontent.com/2234266/79677395-5af9ad80-8223-11ea-8cb7-30617abe77a4.jpeg)

does this antena routing +
 reverse direction M.2 NGFF adaptor method , require to remove parts of the heats spreader ?

**Comment by btbxbob** [comment]
> > It works!
> > attach my photo here.
> > Adjusting the Athena is a little tricky.
> > ![WechatIMG3](https://user-images.githubusercontent.com/2234266/79677392-58975380-8223-11ea-87f5-42986de3da71.jpeg)
> > ![WechatIMG4](https://user-images.githubusercontent.com/2234266/79677394-5a611700-8223-11ea-8500-249057381bfe.jpeg)
> > ![WechatIMG6](https://user-images.githubusercontent.com/2234266/79677395-5af9ad80-8223-11ea-8cb7-30617abe77a4.jpeg)
> 
> does this antena routing +
> reverse direction M.2 NGFF adaptor method , require to remove parts of the heats spreader ?

No, nothing is removed.

**Comment by lf-openthos** [comment]
wow! cool, I am looking for that adaptor now.

Sent from OPENTHOS, an android based open source desktop OS
www.openthos.org


------------------------------------------------------------------

**Comment by SpookHCK** [comment]
https://www.ebay.com/itm/XT-XINTE-Drive-free-BCM943602CS-BCM94360CD-Wireless-Card-to-A-E-Key-Adapter-Card/254657137641

**Comment by daliang1215** [comment]
> It works!
> attach my photo here.
> Adjusting the Athena is a little tricky.
> 
> ![WechatIMG3](https://user-images.githubusercontent.com/2234266/79677392-58975380-8223-11ea-87f5-42986de3da71.jpeg)
> ![WechatIMG4](https://user-images.githubusercontent.com/2234266/79677394-5a611700-8223-11ea-8500-249057381bfe.jpeg)
> ![WechatIMG6](https://user-images.githubusercontent.com/2234266/79677395-5af9ad80-8223-11ea-8cb7-30617abe77a4.jpeg)

Did the wifi signal drop after the reverse card and 2 M.2 cards were installed? On the Internet to see the reverse card from the second M.2 card is too close, heat can not spread very well, resulting in poor wifi/bluetooth.

**Comment by Iaroslav-M** [comment]
> 2 IPEX4 extension

Can you please share your photos of the installation? I still can't get it. Do I need the adapter and BCM943602CS card to avoid blocking the M2 slot and finally use 2 M2 SSD + native Mac WiFi/B card?
Should I buy the IPEX4 extensions, and if so, what for?
I'm very new to all this, and your help will be SO appreciative, please!

**Comment by Iaroslav-M** [comment]
> > > > Alternatively, here's an m.2 card with the macOS supported chip
> > > > [ https://ebay.us/nmM3HK](url)
> > > 
> > > 
> > > could you please provide the card name? link is dead :(
> > 
> > 
> > https://www.ebay.com/itm/Broadcom-BCM94360NG-WiFi-Card-better-than-DW1560-BT4-0-Adapter-802-11ac-1200Mbps/283783627652
> 
> thanks! i bought it

So, this card doesn't require any adapter for use in Haden Canyon, right?

![wireless_card](https://user-images.githubusercontent.com/79906679/109649873-88b2c100-7b6d-11eb-93d8-7a9f13ecdc2c.png)

And what about the space between this wireless card and M2 SSD? Can I use SSD with thermal protection like this?

![adata](https://user-images.githubusercontent.com/79906679/109649899-8e100b80-7b6d-11eb-9b1d-e16ed8fa5def.png)

Will I need additional cable for the antenna?
Thank you for the answer!

**Comment by immie-jay** [comment]
I have the same card - works perfectly. Wifi/BT working.Handoff, Continuity, Apple Watch unlock working. Airdrop as well.
![Screenshot 2021-03-02 at 13 08 46](https://user-images.githubusercontent.com/64142029/109653077-767a5800-7b58-11eb-978e-e2a7e290ac13.png)


Fits perfectly even with XPG GAMMIX S50 

Only difficulty is pushing in the antenna connectors - the inbuilt ones or the longer one with pigtail that is more powerful and sticks onto the lid.

**Reply by keatliang2005** [reply]
this card, come with it's own antenna ? how do you fits the antenna inside the housing
